### PR TITLE
Fix - `azurerm_api_management_named_value` would not enforce `secret=true` when using `value_from_key_vault`

### PR DESCRIPTION
**Pull request by CSymes**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

<br>

## Description

When creating a `azurerm_api_management_named_value` and linking it to a value sourced from a Key Vault via the `value_from_key_vault` block, if the `secret` attribute is not set to `true`, you receive an unexpected 400 error, ala:

```
│ Error: creating or updating Named Value
│ Named Value: "http-function-key"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: ValidationError: One or more fields contain incorrect values:
│ 
│   with azurerm_api_management_named_value.http-function-key,
│   on def_apim.tf line 187, in resource "azurerm_api_management_named_value" "http-function-key":
│  187: resource "azurerm_api_management_named_value" "http-function-key" {
```

This is pretty un-helpful, so this PR adds validation to ensure that these two attributes are defined compatibly.
Unfortunately I couldn't see a way to validate an attribute value based on the state of another attribute, so this is necessarily performed at apply-time, rather than during validation.

This constraint is not well documented, however [MS' examples](https://learn.microsoft.com/en-us/rest/api/apimanagement/named-value/create-or-update?view=rest-apimanagement-2022-08-01&tabs=HTTP#apimanagementcreatenamedvaluewithkeyvault) do display the correct form.

<br>

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass.

## Change Log

* `azurerm_api_management_named_value` - enforce `secret = true` when a `value_from_key_vault` block is defined

This is a (please select all that apply):

- [x] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

## Related Issue(s)
No existing issue that I could see.

### CODE REVIEW

File: internal/services/apimanagement/api_management_named_value_resource.go [hunk 1]
```diff
@@ -129,6 +130,10 @@ func resourceApiManagementNamedValueCreateUpdate(d *pluginsdk.ResourceData, meta
 		},
 	}
 
+	if parameters.Properties.KeyVault != nil && (parameters.Properties.Secret == nil || !*parameters.Properties.Secret) {
+		return fmt.Errorf("secret must be true when value_from_key_vault is set")
```
Comment (stephybun) [inline internal/services/apimanagement/api_management_named_value_resource.go:134]: We usually place property names between back ticks 
```suggestion
		return fmt.Errorf("`secret` must be true when `value_from_key_vault` is set")
```

File: internal/services/apimanagement/api_management_named_value_resource_test.go [hunk 2]
```diff
@@ -63,6 +64,18 @@ func TestAccApiManagementNamedValue_keyVaultSystemAssigned(t *testing.T) {
...
 }
 
+func TestAccApiManagementNamedValue_keyVaultInvalidSecretValue(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_api_management_named_value", "test")
+	r := ApiManagementNamedValueResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config:      r.keyVaultWithInvalidSecretValue(data),
+			ExpectError: regexp.MustCompile(`secret must be true when value_from_key_vault is set`),
```
Comment (stephybun) [inline internal/services/apimanagement/api_management_named_value_resource_test.go:74]:
```suggestion
			ExpectError: regexp.MustCompile(``secret` must be true when `value_from_key_vault` is set`),
```

**Review by stephybun — COMMENTED** [review]
Thanks for adding some validation for this @CSymes! There's one minor comment for the error message for consistency but otherwise this looks good.

**Comment by CSymes** [comment]
Sounds good!

**Comment by CSymes** [comment]
Looks like I've managed to brick the check suite somehow — it was failing on one unit test, so re-ran since that test is fine locally, but I now note that it's likely a main issue as other PRs are failing on the same test, and now a bunch of them are having a bad time, so not sure what's going on there.

**Comment by stephybun** [comment]
@CSymes this should be fixed now. Another rebase on top of main should fix the CI.

**Comment by CSymes** [comment]
@stephybun thanks, all green again.

**Review by katbyte — APPROVED** [review]
Thanks @CSymes - LGTM now 🚜

**Pull request merged**